src/main/java/fr/quatrevieux/araknemu/game/fight/fighter/monster/InvocationFighter.java

```diff
@@ -0,0 +1,78 @@
+package fr.quatrevieux.araknemu.game.fight.fighter.monster;
+
+import java.util.Optional;
+
+import fr.quatrevieux.araknemu.game.fight.castable.weapon.CastableWeapon;
+import fr.quatrevieux.araknemu.game.fight.fighter.AbstractFighter;
+import fr.quatrevieux.araknemu.game.fight.fighter.FighterCharacteristics;
+import fr.quatrevieux.araknemu.game.fight.fighter.FighterLife;
+import fr.quatrevieux.araknemu.game.fight.fighter.PassiveFighter;
+import fr.quatrevieux.araknemu.game.fight.fighter.operation.FighterOperation;
+import fr.quatrevieux.araknemu.game.fight.team.FightTeam;
+import fr.quatrevieux.araknemu.game.spell.SpellList;
+import fr.quatrevieux.araknemu.game.world.creature.Sprite;
+
+final public class InvocationFighter extends AbstractFighter {
+    final private MonsterFighter fighter;
+    final private Optional<PassiveFighter> invoker;
+
+    public InvocationFighter(MonsterFighter fighter, PassiveFighter invoker) {
+        this.fighter = fighter;
+        this.invoker = Optional.of(invoker);
+    }
+
+    @Override
+    public <O extends FighterOperation> O apply(O operation) {
+        return fighter.apply(operation);
```

Comment:
Add method `FighterOperation#onInvocation()` and call it (this is the visitor pattern)
To let the operation decide to apply the operation on the "real" fighter, add a public method for get the fighter instance, with an implementation like :

```java
fighters.apply(new FighterOperation() {
    public void onInvocation(InvocationFighter fighter) {
        fighter.inner().apply(this);
    }

    public void onPlayer(PlayerFighter fighter) {
        // Operation for player fighter
    }
    public void onMonster(MonsterFighter fighter) {
        // Operation for monster
    }
}
```